pyuvdata/mwa_beam.py

```diff
@@ -0,0 +1,573 @@
+            if np.array(freq_range).size != 2:
+                raise ValueError('freq_range must have 2 elements.')
+            freqs_use = freqs_hz[np.nonzero((freqs_hz >= freq_range[0])
+                                            & (freqs_hz <= freq_range[1]))]
+            if freqs_use.size < 1:
+                raise ValueError('No frequencies available in freq_range. '
+                                 'Available frequencies are between {fmin} Hz '
+                                 'and {fmax} Hz'.format(fmin=np.min(freqs_hz),
+                                                        fmax=np.max(freqs_hz)))
+            if freqs_use.size < 2:
+                warnings.warn('Only one available frequency in freq_range.')
+        else:
+            freqs_use = freqs_hz
+
+        beam_modes = self._get_beam_modes(h5filepath, freqs_hz, pol_names,
+                                          dipole_names, max_length, delays,
+                                          amplitudes)
+
+        n_phi = np.floor(360 * pixels_per_deg)
+        n_theta = np.floor(90 * pixels_per_deg) + 1
+        theta_arr = np.deg2rad(np.arange(0, n_theta) / pixels_per_deg)
+        phi_arr = np.deg2rad(np.arange(0, n_phi) / pixels_per_deg)
+
+        jones = self._get_response(freqs_use, pol_names, beam_modes, phi_arr, theta_arr)
+
+        # work out zenith normalization (MWA beams are peak normalized to 1 when pointed at zenith)
+
+        # start filling in the object
+        self.telescope_name = 'MWA'
+        self.feed_name = 'MWA'
+        self.feed_version = '1.0'
+        self.model_name = 'full embedded element'
+        self.model_version = '1.0'
+        self.history = 'Sujito et al. full embedded element beam, derived from https://github.com/MWATelescope/mwa_pb/'
+        if not uvutils._check_history_version(self.history, self.pyuvdata_version_str):
+            self.history += self.pyuvdata_version_str
+
+        self.x_orientation = 'east'
+
+        self.set_efield()
+        self.Naxes_vec = 2
+        self.Ncomponents_vec = 2
+        self.feed_array = np.array([str(pol.lower()) for pol in pol_names])
+        self.Nfeeds = self.feed_array.size
+
+        self.data_normalization = 'physical'
+
+        # for now this returns a simple beam because it requires amps & delays to make the beam
+        self.antenna_type = 'simple'
+
+        self.Nspws = 1
+        self.spw_array = np.array([0])
+        self.Nfreqs = freqs_use.size
+        self.freq_array = freqs_use.astype(np.float)
+        self.freq_array = self.freq_array[np.newaxis, :]
+        self.bandpass_array = np.ones((self.Nspws, self.Nfreqs))
+
+        self.pixel_coordinate_system = 'az_za'
+        self.set_cs_params()
+
+        self.axis1_array = phi_arr
+        self.Naxes1 = self.axis1_array.size
+        self.axis2_array = theta_arr
+        self.Naxes2 = self.axis2_array.size
+
+        self.data_array = np.transpose(jones, axes=[1, 0, 2, 4, 3])
```

Comment:
Can we maybe add a quick comment explaining the input axes, and what they're being transposed to? I think it's useful when permuting quite a few indices like this.

Reply:
yes, good idea.